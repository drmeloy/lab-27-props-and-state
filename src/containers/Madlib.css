.Title {
  width: 75%;
  text-align: center;
  margin: 50px auto;
  font-size: 2rem;
  font-family: 'Montserrat', sans-serif;
}