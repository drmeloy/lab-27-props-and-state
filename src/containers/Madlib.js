import React, { Component } from 'react';
import Form from '../components/madlib/Form';
import Result from '../components/madlib/Result';
import Header from '../components/header/Header';
import Footer from '../components/footer/Footer';
import styles from './Madlib.css';

export default class Madlib extends Component{
  state = {
    showResult: false,
    words: []
  }

  toggleResult = () =>
    this.setState(state => ({ ...state, showResult: !state.showResult }));

  handleSubmit = event => {
    event.preventDefault();
    const elArray = event.target.children;
    let wordsArray = [];

    for (let i = 0; i < elArray.length - 1; i++){
      wordsArray.push(elArray[i].value);
    }

    this.setState({ words: wordsArray });
    this.toggleResult();
  }

  clearForm = event => {
    event.preventDefault();
    const elArray = event.target.children;
    for (let i = 0; i < elArray.length - 1; i++){
      elArray[i].value = '';
    }
  }

  render() {
    const { showResult, words } = this.state;
    
    return (
      <>
        <Header />
        <p className={styles.Title}>Fill out this MadLib for a fun story!</p>
        {!showResult && <Form onSubmit={this.handleSubmit} />}
        {showResult && <Result words={words} closeResult={this.toggleResult} />}
        <Footer />
      </>
    );
  }
}
