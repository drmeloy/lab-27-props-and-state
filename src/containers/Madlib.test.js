import React from 'react';
import { shallow } from 'enzyme';
import Madlib from './Madlib';

describe('Madlib component', () => {
  it('renders Madlib', () => {
    const wrapper = shallow(<Madlib />);
    expect(wrapper).toMatchSnapshot();
  });
});