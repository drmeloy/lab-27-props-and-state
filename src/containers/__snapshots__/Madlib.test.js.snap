// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Madlib component renders Madlib 1`] = `
<Fragment>
  <Header />
  <p
    className="Title"
  >
    Fill out this MadLib for a fun story!
  </p>
  <Form
    onSubmit={[Function]}
  />
  <Footer />
</Fragment>
`;
